app/data_client.go

```diff
@@ -471,6 +471,12 @@ func newDataClient(conn rpc.ClientConn) *DataClient {
 	}
 }
 
+// CreateDataClientWithDataServiceClient creates a DataClient with a manually set DataServiceClient.
+// This is needed for unit tests and should not be used elsewhere.
```

Comment:
just spitballing here, I need to think on it more so no specific ask/request at this point, but: while I agree that this needs to be public, I don't love the comment saying "should not be used elsewhere." It just feels kinda strange. I think we could say "Public for tests" (a phrase that shows up in the codebase a fair amount), or we could probably leave it as-is? I don't think there's any real harm in allowing this to exist in which case it's fine to let people use it. And if I'm wrong and there _is_ potential risk then we'll need a better solution than saying "please don't abuse this". 

@n0nick @katiepeters do y'all have thoughts about this being public?